test/stdlib/ValueEnumerableTests.swift

```diff
@@ -0,0 +1,76 @@
+// RUN: %target-run-simple-swift %t
+// REQUIRES: executable_test
+
+import StdlibUnittest
+
+var ValueEnumerableTests = TestSuite("ValueEnumerableTests")
+
+enum InDeclaration: ValueEnumerable {
+  case bar
+  case baz
+  case quux
+}
+
+enum InExtension {
+  case quuz
+  case corge
+  case grault
+  case garply
+}
+
+extension InExtension: ValueEnumerable {}
```

Comment:
We should probably have test cases to show that extensions can’t be used to derive a conformance for an enum from another module or imported from a header.

Reply:
The proposal wasn't clear on this.  Equatable, Hashable, and Codable all disallow synthesis in extensions even for the same module.  If that's the desired behavior I can remove this test.